### PR TITLE
S3 deploy

### PR DESCRIPTION
**Pull request by farski**
This works, but I'm guessing I didn't do as well as I should have.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,7 +1,89 @@
...
+  return gulp
+    .src('./index.jade')
+    .pipe(jade({ locals: YOUR_LOCALS }))
+    .pipe(gulp.dest('./.dist'));
+});
+
+gulp.task('jspmBundleSfx', function (callback) {
+  return gulp
+    .src('./app/main.ts')
+    .pipe(jspm({ selfExecutingBundle: true, minify: true, mangle: false }))
```
Comment (cavis) [inline gulpfile.js]: Nice to have: use `gulp-sourcemaps` around this.  The `gulp-jspm` docs say they support it.

File: gulpfile.js [hunk 2]
```diff
@@ -1,7 +1,89 @@
...
+});
+
+gulp.task('copyPolyfills', function (callback) {
+  return gulp
+    .src('./node_modules/angular2/bundles/angular2-polyfills.js')
+    .pipe(gulp.dest('./.dist/scripts'));
+});
+
+gulp.task('uploadToS3', function (callback) {
+  aws = JSON.parse(fs.readFileSync('./aws.json'));
```
Comment (cavis) [inline gulpfile.js:65]: I find this a little wonky.  But it looks like `gulp-s3` doesn't support the usual IAM/`~/.aws/credentials`/ENV-variable credentials.  Lame!

Comment (farski) [inline gulpfile.js:65]: I don't think the S3 situation is permanent, anyway.

**Comment by cavis** [comment]
Looks good to me.  Adding sourcemaps is optional ... so do you want to add that to this PR, or shall I merge?

File: gulpfile.js (hunk 1 above)
Comment (farski) [inline gulpfile.js]: Adding that increased the size of the deployable library by almost 3 MB. `jspm()` never gets run in development really, so does it make sense to have the sourcemaps?

Comment (cavis) [inline gulpfile.js]: Are you writing the sourcemaps back into `backsaw.min.js`?  It should put them in an external `backsaw.min.js.map` file by doing `sourcemaps.write('.')`.

Comment (farski) [inline gulpfile.js]: Ah. And I want that to go out with the deploy?

**Comment by farski** [comment]
Added the source maps, so should be good to merge

File: gulpfile.js (hunk 1 above)
Comment (cavis) [inline gulpfile.js]: Yeah, I think so.  Browsers will only download the map file if the devtools are open.

**Comment by cavis** [comment]
:ship: :it:

**Pull request merged**